openeo/metadata.py

```diff
@@ -522,3 +523,61 @@ def _repr_html_(self):
     def __str__(self) -> str:
         bands = self.band_names if self.has_band_dimension() else "no bands dimension"
         return f"CollectionMetadata({self.extent} - {bands} - {self.dimension_names()})"
+
+
+def metadata_from_stac(url: str) -> CubeMetadata:
+    """
+    Reads the band metadata a static STAC catalog or a STAC API Collection and returns it as a :py:class:`CubeMetadata`
+
+    :param url: The URL to a static STAC catalog (STAC Item, STAC Collection, or STAC Catalog) or a specific STAC API Collection
+    :return: A :py:class:`CubeMetadata` containing the DataCube band metadata from the url.
+    """
+
+    def get_band_names(asst: pystac.Asset) -> List[Band]:
+        return [Band(eo_band["name"]) for eo_band in asst.extra_fields["eo:bands"]]
+
+    def is_band_asset(asset: pystac.Asset) -> bool:
+        return "eo:bands" in asset.extra_fields
+
+    stac_object = pystac.read_file(href=url)
+
+    band_names = []
+    collection = None
+
+    if isinstance(stac_object, pystac.Item):
+        item = stac_object
+        if "eo:bands" in item.properties:
+            eo_bands_location = item.properties
+        elif item.get_collection() is not None:
+            collection = item.get_collection()
+            eo_bands_location = item.get_collection().summaries.lists
+        else:
+            eo_bands_location = {}
+        band_names = [Band(b["name"]) for b in eo_bands_location.get("eo:bands", [])]
+
+    elif isinstance(stac_object, pystac.Collection):
+        collection = stac_object
+        band_names = [Band(b["name"]) for b in collection.summaries.lists.get("eo:bands", [])]
+
+        # Summaries is not a required field in a STAC collection, so also check the assets
+        for itm in collection.get_items():
+            band_assets = {
+                asset_id: asset
+                for asset_id, asset in dict(sorted(itm.get_assets().items())).items()
```

Comment:
why sorting here? isn't that going to mix up existing metadata orders?

Reply:
This does not sort the band names, but rather the assets in the item. In the backend, the assets are also sorted before the band metadata is read, so I included it here as well, to make sure there is no mismatch between the backend and client implementation.
https://github.com/Open-EO/openeo-geopyspark-driver/blob/7caaa9aa74585bad3c42e054cfb5dfe95ed31b55/openeogeotrellis/backend.py#L1003

Reply:
The openeo client should be usable with other backends than ours, so we should avoid hardcoding geopyspark behavior like that. And even for the geopyspark backend it's questioned if we should do that kind of sorting.
e.g. see https://github.com/Open-EO/openeo-processes/issues/488